Application/EdFi.Ods.Extensions.TPDM/Artifacts/MsSql/Data/Security/1010-TPDM-ResourceClaims.sql

```diff
@@ -1,465 +1,6004 @@
-);
+BEGIN
+    DECLARE 
+        @applicationId AS INT,
+        @claimId AS INT,
+        @claimName AS nvarchar(max),
+        @parentResourceClaimId AS INT,
+        @existingParentResourceClaimId AS INT,
+        @claimSetId AS INT,
+        @claimSetName AS nvarchar(max),
+        @authorizationStrategyId AS INT,
+        @msg AS nvarchar(max),
+        @createActionId AS INT,
+        @readActionId AS INT,
+        @updateActionId AS INT,
+        @deleteActionId AS INT
+
+    DECLARE @claimIdStack AS TABLE (Id INT IDENTITY, ResourceClaimId INT)
+
+    SELECT @applicationId = ApplicationId
+    FROM [dbo].[Applications] WHERE ApplicationName = 'Ed-Fi ODS API';
+
+    SELECT @createActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Create';
+
+    SELECT @readActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Read';
+
+    SELECT @updateActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Update';
+
+    SELECT @deleteActionId = ActionId
+    FROM [dbo].[Actions] WHERE ActionName = 'Delete';
+
+    BEGIN TRANSACTION
+
+    -- Push claimId to the stack
+    INSERT INTO @claimIdStack (ResourceClaimId) VALUES (@claimId)
+
+    -- Processing children of root
+    ----------------------------------------------------------------------------------------------------------------------------
+    -- Resource Claim: 'http://ed-fi.org/ods/identity/claims/domains/tpdm'
+    ----------------------------------------------------------------------------------------------------------------------------
+    SET @claimName = 'http://ed-fi.org/ods/identity/claims/domains/tpdm'
+    SET @claimId = NULL
+
+    SELECT @claimId = ResourceClaimId, @existingParentResourceClaimId = ParentResourceClaimId
+    FROM dbo.ResourceClaims 
+    WHERE ClaimName = @claimName
+
+    SELECT @parentResourceClaimId = ResourceClaimId
+    FROM @claimIdStack
+    WHERE Id = (SELECT Max(Id) FROM @claimIdStack)
+
+    IF @claimId IS NULL
+        BEGIN
+            PRINT 'Creating new claim: ' + @claimName
+
+            INSERT INTO dbo.ResourceClaims(DisplayName, ResourceName, ClaimName, ParentResourceClaimId, Application_ApplicationId)
+            VALUES ('tpdm', 'tpdm', 'http://ed-fi.org/ods/identity/claims/domains/tpdm', @parentResourceClaimId, @applicationId)
+
+            SET @claimId = SCOPE_IDENTITY()
+        END
+    ELSE
+        BEGIN
+            IF @parentResourceClaimId != @existingParentResourceClaimId
+            BEGIN
+                PRINT 'Repointing claim ''' + @claimName + ''' (ResourceClaimId=' + CONVERT(nvarchar, @claimId) + ') to new parent (ResourceClaimId=' + CONVERT(nvarchar, @parentResourceClaimId) + ')'
+
+                UPDATE dbo.ResourceClaims
+                SET ParentResourceClaimId = @parentResourceClaimId
+                WHERE ResourceClaimId = @claimId
+            END
+        END
+  
+    -- Processing claim sets for http://ed-fi.org/ods/identity/claims/domains/tpdm
```

Comment:
That statement is sort of true, but I covered it in my earlier response. It depends on how the script being produced is inteded to be used. The use case I started with was producing an additional script to fix up (rather than completely replace) the existing metadata. Even with a full replacement of the 1010 TPDM security metadata script, there's still the need to reference the existing Ed-Fi security metadata (particularly for expanding on the existing claims hierarchy) without fear of overwriting what's there (or inadvertently changing it if it had been customized). That was my approach here.